sector-builder/src/builder.rs

```diff
@@ -26,39 +32,23 @@ pub struct SectorBuilder {
 
     // The main worker. Owns all mutable state for the SectorBuilder.
     scheduler: Scheduler,
-
-    // Configures size of proofs and sectors managed by the SectorBuilder.
-    sector_class: SectorClass,
 }
 
 impl SectorBuilder {
     // Initialize and return a SectorBuilder from metadata persisted to disk if
     // it exists. Otherwise, initialize and return a fresh SectorBuilder. The
     // metadata key is equal to the prover_id.
-    pub fn init_from_metadata<S: Into<String>>(
+    pub fn init_from_metadata(
         sector_class: SectorClass,
         last_committed_sector_id: SectorId,
-        metadata_dir: S,
+        metadata_dir: String,
         prover_id: [u8; 31],
-        sealed_sector_dir: S,
-        staged_sector_dir: S,
+        sealed_sector_dir: String,
+        staged_sector_dir: String,
```

Comment:
Nit: directories should ideally be `impl AsRef<Path>`

Reply:
and stored as `PathBuf`

Reply:
good call. updated.